### PR TITLE
Passthrough failure mode allow 1.12 backport

### PR DESCRIPTION
**Pull request by inFocus7**
_backport of #6942 + #7314_

# Description

- Added `failure_mode_allow` to the `PassThroughAuth` definition.

# Context

We want something similar to the global `failure_mode_allow` setting, just for our passthrough auth. 

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [X] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by inFocus7** [comment]
adding WIP to this to cherry-pick teh small doc updates here https://github.com/solo-io/gloo/pull/7314

**Review by jackstine — DISMISSED** [review]

**Review by sam-heilbron — CHANGES_REQUESTED** [review]
Let's pull in the updates from https://github.com/solo-io/gloo/pull/7314 (for guides, api comments and metrics)

**Comment by sam-heilbron** [comment]
This looks good to me. Leaving open so @elcasteel and @jackstine can review

**Review by jackstine — APPROVED** [review]

File: projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto [hunk 1]
```diff
@@ -1451,6 +1455,7 @@ message ExtAuthConfig {
     // How to handle processing of named configs within an auth config chain.
     // An example config might be: `( basic1 || basic2 || (oidc1 && !oidc2) )`
     // The boolean expression is evaluated left to right but honors parenthesis and short-circuiting.
+    // Defaults to an empty string, which is interpreted as `and`-ing the configs.
```
Comment (kdorosh) [inline projects/gloo/api/v1/enterprise/options/extauth/v1/extauth.proto:1458]: :clap:

**Pull request merged**